File: scripts/take_screenshots.py
#!/usr/bin/env python3
"""Take real screenshots of ParkHub using Selenium + Firefox."""
import time
import os
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.options import Options
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

os.makedirs("docs/screenshots", exist_ok=True)

opts = Options()
opts.add_argument("--headless")
opts.add_argument("--width=1280")
opts.add_argument("--height=800")
opts.set_preference("ui.systemUsesDarkTheme", 1)

driver = webdriver.Firefox(options=opts)
driver.set_window_size(1280, 800)
BASE = "http://localhost:7878"

def shot(name, delay=2):
    time.sleep(delay)
    path = f"docs/screenshots/{name}.png"
    driver.save_screenshot(path)
    print(f"✓ {path}")

# Welcome (unauthenticated)
driver.get(f"{BASE}/welcome")
shot("welcome")

# Login page
driver.get(f"{BASE}/login")
shot("login")

# Perform login
try:
    inputs = driver.find_elements(By.CSS_SELECTOR, "input")
    if len(inputs) >= 2:
        inputs[0].clear()
        inputs[0].send_keys("admin")
        inputs[1].clear()
        inputs[1].send_keys("Test1234!")
    submit = driver.find_element(By.CSS_SELECTOR, "button[type='submit']")
    submit.click()
    time.sleep(3)
except Exception as e:
    print(f"Login attempt: {e}")

# Dashboard
shot("dashboard")

# Booking
driver.get(f"{BASE}/book")
shot("booking")

# Admin
driver.get(f"{BASE}/admin")
shot("admin")

# Settings/themes
driver.get(f"{BASE}/settings")
shot("themes")

# Mobile view
driver.set_window_size(375, 812)
driver.get(f"{BASE}/")
shot("mobile")

# Dark mode booking mobile
driver.get(f"{BASE}/book")
shot("dark-mode")

driver.quit()
print("\nAll screenshots captured!")

File: scripts/take_screenshots.js
const { chromium, firefox } = require('playwright');

(async () => {
  let browser;
  try {
    browser = await chromium.launch({ headless: true });
    console.log('Using Chromium');
  } catch (e) {
    console.log('Chromium failed, trying Firefox...', e.message);
    try {
      browser = await firefox.launch({ headless: true });
      console.log('Using Firefox');
    } catch (e2) {
      console.error('Both browsers failed:', e2.message);
      process.exit(1);
    }
  }

  const context = await browser.newContext({
    viewport: { width: 1280, height: 800 },
    colorScheme: 'dark'
  });
  const page = await context.newPage();

  // Login page
  await page.goto('http://localhost:7878/login');
  await page.waitForTimeout(2000);
  await page.screenshot({ path: 'docs/screenshots/login.png' });
  console.log('✓ login.png');

  // Login
  const usernameInput = page.locator('input').first();
  await usernameInput.fill('admin');
  const passwordInput = page.locator('input[type="password"]');
  await passwordInput.fill('Test1234!');
  await page.locator('button[type="submit"]').click();
  await page.waitForTimeout(2000);

  // Dashboard
  await page.screenshot({ path: 'docs/screenshots/dashboard.png' });
  console.log('✓ dashboard.png');

  // Booking page
  await page.goto('http://localhost:7878/book');
  await page.waitForTimeout(2000);
  await page.screenshot({ path: 'docs/screenshots/booking.png' });
  console.log('✓ booking.png');

  // Admin page
  await page.goto('http://localhost:7878/admin');
  await page.waitForTimeout(2000);
  await page.screenshot({ path: 'docs/screenshots/admin.png' });
  console.log('✓ admin.png');

  // Settings/themes
  await page.goto('http://localhost:7878/settings');
  await page.waitForTimeout(1500);
  await page.screenshot({ path: 'docs/screenshots/themes.png' });
  console.log('✓ themes.png');

  // Welcome page (unauthenticated view)
  const context2 = await browser.newContext({
    viewport: { width: 1280, height: 800 },
    colorScheme: 'dark'
  });
  const page2 = await context2.newPage();
  await page2.goto('http://localhost:7878/welcome');
  await page2.waitForTimeout(2000);
  await page2.screenshot({ path: 'docs/screenshots/welcome.png' });
  console.log('✓ welcome.png');
  await context2.close();

  // Mobile view
  await page.setViewportSize({ width: 375, height: 812 });
  await page.goto('http://localhost:7878/');
  await page.waitForTimeout(2000);
  await page.screenshot({ path: 'docs/screenshots/mobile.png' });
  console.log('✓ mobile.png');

  // Dark mode booking (mobile)
  await page.goto('http://localhost:7878/book');
  await page.waitForTimeout(2000);
  await page.screenshot({ path: 'docs/screenshots/dark-mode.png' });
  console.log('✓ dark-mode.png');

  await browser.close();
  console.log('\nAll screenshots captured!');
})();
